samples/apps/video-streaming-app/streaming.py

```diff
@@ -16,54 +16,47 @@
 
 from flask import Flask, render_template, Response
 
-camera_frame_queues = []
+from kubernetes import client, config
+import re
 
-main_frame_source = ""
+camera_frame_queues = []
 small_frame_sources = []
 
-if 'CONFIGURATION_NAME' in os.environ:
-    configuration_name = os.environ['CONFIGURATION_NAME']
-    short_env_var_prefix = (configuration_name + '-').upper().replace('-', '_')
-    # For every k8s service, an env var is set in every node with its ports.  The
-    # format is <SERVICE_NAME_IN_CAPS>_SERVICE_PORT_<PORT_NAME>.  Here, we will query
-    # these values on the streaming app node, to get the exposed port number ... but
-    # we will assume that the ports are named 'grpc', if the name changes, THIS CODE
-    # WILL BREAK.
-    env_var_prefix = short_env_var_prefix + 'SVC_SERVICE_'
-    grpc_port = os.environ[env_var_prefix + 'PORT_GRPC'] # instance services are using the same port by default
-    main_frame_source = "{0}:{1}".format(os.environ[env_var_prefix + 'HOST'], grpc_port)
-    instance_service_hosts = filter(
-        lambda name: name.startswith(short_env_var_prefix) and not name.startswith(env_var_prefix) and name.endswith('_SERVICE_HOST'),
-        os.environ)
-    camera_count = 0
-    for svc_host_env_var in instance_service_hosts:
-        url = "{0}:{1}".format(os.environ[svc_host_env_var], grpc_port)
-        small_frame_sources.append(url)
-        camera_count += 1
-
-else:
-    camera_count = int(os.environ['CAMERA_COUNT'])
-    main_frame_source = "{0}:80".format(os.environ['CAMERAS_SOURCE_SVC'])
-    for camera_id in range(1, camera_count + 1):
-        url = "{0}:80".format(os.environ['CAMERA{0}_SOURCE_SVC'.format(camera_id)])
-        small_frame_sources.append(url)
-
-for camera_id in range(camera_count + 1):
-    camera_frame_queues.append(queue.Queue(1))
+def get_camera_list(configuration_name):
+    camera_list = []
+    config.load_incluster_config()
+    coreV1Api = client.CoreV1Api()

```

Comment:
Couldn't we query the env vars set on the node periodically for services (like in the previous implementation) instead of connecting to the K8s API?

Reply:
a node's env vars are not updated.

Reply:
Is it that the container only receives the state of the nodes env vars when first deployed?

Reply:
i'd be speculating on the hows or the whys, but I found that if you:

1. start serviceA
1. start podA, it has serviceA details in its env vars
1. start serviceB
1. start podB, it has serviceA and serviceB details in env vars
1. podA never gets serviceB details in env vars

Reply:
My understanding is that the envs reflect only the state at the start of the pod. That makes sense, as the env vars and injected into the process, no way to modify after the process starts.